packages/cloud/connection_details/components/fancy_select/fancy_select.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as React from 'react';
+import {
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFieldText,
+  EuiFormControlLayout,
+  EuiHorizontalRule,
+  EuiInputPopover,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import type { FancySelectOption } from './types';
+
+export interface FancySelectProps {
+  value: string;
+  options: FancySelectOption[];
+  onChange: (value: string) => void;
+}
+
+export const FancySelect: React.FC<FancySelectProps> = ({ value, options, onChange }) => {
```

Comment:
yes :( while testing noticed that even navigation from the keyboard doesn't work (don't know about screenreader)

Reply:
Because I cannot get the styling as per Figma with the `EuiSuperSelect`. But I agree it is probably not worth it and accessibility will be better. I've moved code to use `EuiSuperSelect`, the spacing around icons is a bit off, but I still think it looks good.